src/buddyinfo.c

```diff
@@ -0,0 +1,130 @@
+/**
+ * collectd - src/buddyinfo.c
+ * Copyright (C) 2019       Asaf Kahlon
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; only version 2 of the License is applicable.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * Authors:
+ *   Asaf Kahlon <asafka7 at gmail.com>
+ **/
+
+#include "collectd.h"
+
+#include "plugin.h"
+#include "utils/common/common.h"
+#include "utils/ignorelist/ignorelist.h"
+
+#if !KERNEL_LINUX
+#error "No applicable input method."
+#endif
+
+#include <unistd.h>
+
+#define MAX_ORDER 11
+#define BUDDYINFO_FIELDS MAX_ORDER + 2 // "zone" + Name + (MAX_ORDER entries)
+#define NUM_OF_KB(pagesize, order) ((pagesize) / 1024) * (1 << (order))
+
+static const char *config_keys[] = {"Zone"};
+static int config_keys_num = STATIC_ARRAY_SIZE(config_keys);
+
+static ignorelist_t *ignorelist;
+
+static int buddyinfo_config(const char *key, const char *value) {
+  if (ignorelist == NULL) {
+    ignorelist = ignorelist_create(1);
+    if (ignorelist != NULL) {
```

Comment:
if (ignorelist == NULL) {
  ERROR("buddyinfo plugin: ignorelist_create failed");
  return -ENOMEM; // be consistent  and return negative numbers on error
}

Please test your commits before submitting. In earlier comment I refered to (ignorelist != NULL) as a positive case.